src/ipc/client.rs

```diff
@@ -260,6 +261,25 @@ pub fn handle_msg(msg: Msg, json: bool) -> anyhow::Result<()> {
                 println!("The change will apply when it is connected.");
             }
         }
+        Msg::Workspaces => {
+            let Response::Workspaces(response) = response else {
+                bail!("unexpected response: expected Workspaces, got {response:?}");
+            };
+
+            if json {
+                let response =
+                    serde_json::to_string(&response).context("error formatting response")?;
+                println!("{response}");
+                return Ok(());
+            }
+
+            for workspace in response {
+                let display = workspace.monitor;
+                let active = if workspace.active { " *" } else { "" };
+                let workspace = workspace.id;
+                println!("{display}: {workspace}{active}");
+            }
```

Comment:
This should sort and display workspaces per monitor. If you don't mind, I will just change this to my liking myself after the other fixes.

Reply:
Totally fine with me, have at it.